### DISCUSSION TITLE
SupportDisablingBackButtonCache::$disableBackButtonCache; Static variable poor design and bug

### DISCUSSION THREAD

**Discussion by AidasK** [livewire/livewire#8198] [7. Bugs (archived)]
# The issue
Livewire does modify cache headers if  `SupportDisablingBackButtonCache::$disableBackButtonCache` is set to true. But the problem is that this variable is not being reset after the requests. So Tests are failing which are running after livewire tests. 

Livewire should not use static variable and instead it should use some laravel singleton class or it should reset the state after the request.

Probably same issue happens for Octane apps and it overwrites all cache headers which might lead to terrible cache issues.

https://github.com/livewire/livewire/blob/e62eb896f6b03f78c8aade9f85ec61dbe075a7f4/src/Features/SupportDisablingBackButtonCache/HandlesDisablingBackButtonCache.php#L9

`SupportDisablingBackButtonCache::$disableBackButtonCache` variable is only set to true, but no code part sets it to false. 

Probably this middleware after the response could reset the state back?
https://github.com/livewire/livewire/blob/e62eb896f6b03f78c8aade9f85ec61dbe075a7f4/src/Features/SupportDisablingBackButtonCache/DisableBackButtonCacheMiddleware.php#L21

# How to reproduce this bug:
1. Write some test case which validated response cache headers, something like 
    $response->assertHeader('Cache-Control', 'no-cache, no-store, private');
2. Write some livewire tests as well.
3. Independently all tests pass. But running all together header test fails as livewire alters the headers and does not reset the state.

# Additional info

This should have been a bug report as it causes our tests to fail randomly, but sorry as I am being too lazy to make a repository for reproduction.

Other parts of the code also suffers because of this issue:
https://github.com/livewire/livewire/blob/e62eb896f6b03f78c8aade9f85ec61dbe075a7f4/src/Features/SupportFileUploads/UnitTest.php#L469

**Comment by joshhanley** [comment]
@AidasK yeah looks like a bug to me. Have moved this to the bug category. PRs to fix are welcome!

**Comment by NebsterSK** [comment]
I need to customize the same 3 response headers that `DisableBackButtonCacheMiddleware.php` changes because of internal security request, but **Livewire overwrites it**. Changing middleware order in `Kernel.php` did not help. 

Hotfix (not really nice solution) is putting this code in `AppServiceProvider.php`

```
Livewire::listen('mount', function ($component) {
    $component->enableBackButtonCache();
});
```
        
 It would be nice to be able to disable this feature or to customize the headers in config.

**Reply by fabricioffa** [reply]
I tried this today i the update requests still have the headers. The only work around for me was to create a middleware with the same name and path and load it instead of the original one. But I'm not willing to push this thing to prod

**Reply by thomaspalmer** [reply]
Chasing my tail for hours! Thank you sir, hopefully a proper solution will be available though.

**Comment by joshhanley** [comment]
I've just submitted a PR #9327 with a fix for this.
